packages/admin-next/dashboard/src/v2-routes/products/common/components/category-combobox/category-combobox.tsx

```diff
@@ -0,0 +1,380 @@
+  > {
+  value: string[]
+  onChange: (value: string[]) => void
+}
+
+type Level = {
+  id: string
+  label: string
+}
+
+export const CategoryCombobox = forwardRef<
+  HTMLInputElement,
+  CategoryComboboxProps
+>(({ value, onChange, className, ...props }, ref) => {
+  const innerRef = useRef<HTMLInputElement>(null)
+
+  useImperativeHandle<HTMLInputElement | null, HTMLInputElement | null>(
+    ref,
+    () => innerRef.current,
+    []
+  )
+
+  const [open, setOpen] = useState(false)
+
+  const { i18n, t } = useTranslation()
+
+  const [level, setLevel] = useState<Level[]>([])
+  const { searchValue, onSearchValueChange, query } = useDebouncedSearch()
+
+  const { product_categories, isPending, isError, error } = useCategories(
+    {
+      q: query,
+      parent_category_id: !searchValue ? getParentId(level) : undefined,
+      include_descendants_tree: !searchValue ? true : false,
+    },
+    {
+      enabled: open,
+    }
+  )
+
+  const [showLoading, setShowLoading] = useState(false)
+
+  /**
+   * We add a small artificial delay to the end of the loading state,
+   * this is done to prevent the popover from flickering too much when
+   * navigating between levels or searching.
+   */
+  useEffect(() => {
+    let timeoutId: ReturnType<typeof setTimeout> | undefined
+
+    if (isPending) {
+      setShowLoading(true)
+    } else {
+      timeoutId = setTimeout(() => {
+        setShowLoading(false)
+      }, 150)
+    }
+
+    return () => {
+      clearTimeout(timeoutId)
+    }
+  }, [isPending])
+
+  useEffect(() => {
+    if (searchValue) {
+      setLevel([])
+    }
+  }, [searchValue])
+
+  function handleLevelUp(e: MouseEvent<HTMLButtonElement>) {
+    e.preventDefault()
+    e.stopPropagation()
+
+    setLevel(level.slice(0, level.length - 1))
+
+    innerRef.current?.focus()
+  }
+
+  function handleLevelDown(option: ProductCategoryOption) {
+    return (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      setLevel([...level, { id: option.value, label: option.label }])
+
+      innerRef.current?.focus()
+    }
+  }
+
+  function handleSelect(option: ProductCategoryOption) {
+    return (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      if (isSelected(value, option.value)) {
+        onChange(value.filter((v) => v !== option.value))
+      } else {
+        onChange([...value, option.value])
+      }
+
+      innerRef.current?.focus()
+    }
+  }
+
+  function handleOpenChange(open: boolean) {
+    if (!open) {
+      onSearchValueChange("")
+      setLevel([])
+    }
+
+    if (open) {
+      requestAnimationFrame(() => {
+        innerRef.current?.focus()
+      })
+    }
+
+    setOpen(open)
+  }
+
+  const options = getOptions(product_categories || [])
+
+  const showTag = value.length > 0 && !open
+
+  if (isError) {
+    throw error
+  }
+
+  return (
+    <Popover.Root open={open} onOpenChange={handleOpenChange}>
```

Comment:
nit: when this component is near the bottom of the screen, when you type, the popover sometimes renders above the input while it loads and then is rendered below the input when the fetch is over. Can we force the popover to always render to the bottom of the input?